src/validation_utils.py

```diff
@@ -13,32 +18,69 @@ def type_is_equal(col_type: str, val_type: str) -> bool:
         return True
     return False
 
-def check_column_types(data, validation):
-    """Iterate over all csv files and check whether columns are well formatted."""
+def check_column_types(data: dict, validation: dict) -> list:
+    """Iterate over all csv files and check whether columns are well formatted.
+
+    Parameters
+    ----------
+    data:
+        Dictionary mapping from filename to Pandas.DataFrame
+    validation:
+        Dictionary rendered from the validation toml.
+
+    Returns
+    -------
+    List of tuples [(data_name, column_name, found_type, expected_type)]
```

Comment:
Only failed columns are returned, right? Then that should be cleared up in the docstrings.